staging/src/k8s.io/legacy-cloud-providers/gce/gce_instances.go

```diff
@@ -119,19 +172,32 @@ func (g *Cloud) NodeAddressesByProviderID(ctx context.Context, providerID string
 		return []v1.NodeAddress{}, err
 	}
 
-	instance, err := g.c.Instances().Get(timeoutCtx, meta.ZonalKey(canonicalizeInstanceName(name), zone))
+	nodeAddresses, err := g.NodeAddressesByInstanceNameZone(timeoutCtx, name, zone)
+
+	return nodeAddresses, err
+
+}
+
+// NodeAddressesByInstanceNameZone uses the GCE API to get the internal and external IPs
+// from all the interfaces on the VM
+func (g *Cloud) NodeAddressesByInstanceNameZone(ctx context.Context, instanceName string, zone string) ([]v1.NodeAddress, error) {
```

Comment:
this function should be private, as it did not implement anything from `Instances`, and it is purely something internal to gcp cloud provider (potential user of gcp cloud provider should not worry about zone, etc).